### PR TITLE
Support for dns `multiaddrs`

### LINKED ISSUE #521: Does not support `dns4` multiaddr
From 7e9dac67c8927c953687ee6391c63286377bb9c9

It looks like it does not support `dns4` multiaddr. I get the following error when connecting to a go-waku node using domain name:

```
▶ ./build/chat2 --staticnode:/dns4/node-01.do-ams3.jdev.misc.statusim.net/tcp/9000/p2p/16Uiu2HAm7gwDpb2SQkWv69V7YND1v9tzs12eu1YyqNYmtEZwvC8D       
Choose a nickname >> fran
Welcome, fran!
Connecting to nodes
libbacktrace error: no debugging symbols available. Compile with '--debugger:native'.
libbacktrace error: no debugging symbols available. Compile with '--debugger:native'.
libbacktrace error: no debugging symbols available. Compile with '--debugger:native'.
libbacktrace error: no debugging symbols available. Compile with '--debugger:native'.
libbacktrace error: no debugging symbols available. Compile with '--debugger:native'.
libbacktrace error: no debugging symbols available. Compile with '--debugger:native'.
/home/froyer/src/status-im/nim-waku/waku/v2/node/storage/peer/peer_storage.nim(263) main
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(256) NimMain
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(1092) waitFor
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(281) poll
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(74) connectToNodes
/home/froyer/src/status-im/nim-waku/examples/v2/chat2.nim(262) processInput
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(101) colonanonymous
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(74) colonanonymous
/home/froyer/src/status-im/nim-waku/examples/v2/chat2.nim(110) connectToNodes
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(101) colonanonymous
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(74) processInput
/home/froyer/src/status-im/nim-waku/waku/v2/node/wakunode2.nim(505) processInput
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(101) connectToNodes
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncloop.nim(74) connectToNodes
/home/froyer/src/status-im/nim-waku/waku/v2/node/wakunode2.nim(480) colonanonymous
/home/froyer/src/status-im/nim-waku/waku/v2/utils/peers.nim(38) colonanonymous
[[reraised from:
/home/froyer/src/status-im/nim-waku/waku/v2/node/storage/peer/peer_storage.nim(263) main
/home/froyer/src/status-im/nim-waku/waku/v2/node/storage/peer/peer_storage.nim(256) NimMain
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncloop.nim(1092) waitFor
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncloop.nim(281) poll
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(74) colonanonymous
/home/froyer/src/status-im/nim-waku/examples/v2/chat2.nim(262) processInput
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(101) connectToNodes
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(74) colonanonymous
/home/froyer/src/status-im/nim-waku/examples/v2/chat2.nim(110) connectToNodes
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(101) connectToNodes
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(77) colonanonymous
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(367) connectToNodes
]]
[[reraised from:
/home/froyer/src/status-im/nim-waku/waku/v2/node/storage/peer/peer_storage.nim(263) main
/home/froyer/src/status-im/nim-waku/waku/v2/node/storage/peer/peer_storage.nim(256) NimMain
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncloop.nim(1092) waitFor
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncloop.nim(281) poll
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(74) colonanonymous
/home/froyer/src/status-im/nim-waku/examples/v2/chat2.nim(262) processInput
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(101) connectToNodes
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(77) colonanonymous
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(367) connectToNodes
]]
[[reraised from:
/home/froyer/src/status-im/nim-waku/waku/v2/node/storage/peer/peer_storage.nim(263) main
/home/froyer/src/status-im/nim-waku/waku/v2/node/storage/peer/peer_storage.nim(256) NimMain
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncloop.nim(1092) waitFor
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncloop.nim(281) poll
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(77) colonanonymous
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(367) processInput
]]
[[reraised from:
/home/froyer/src/status-im/nim-waku/waku/v2/node/storage/peer/peer_storage.nim(263) main
/home/froyer/src/status-im/nim-waku/waku/v2/node/storage/peer/peer_storage.nim(256) NimMain
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncloop.nim(1092) waitFor
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncloop.nim(281) poll
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(74) colonanonymous
/home/froyer/src/status-im/nim-waku/vendor/nim-chronos/chronos/asyncmacro2.nim(367) main
]]
[[reraised from:
/home/froyer/src/status-im/nim-waku/waku/v2/node/storage/peer/peer_storage.nim(263) main
/home/froyer/src/status-im/nim-waku/waku/v2/node/storage/peer/peer_storage.nim(256) NimMain
]]
Error: unhandled exception: Invalid node multi-address [ValueError]
```

No problem when using ip.

### PR DESCRIPTION
**Pull request by jm-clius**
This PR closes #521

It adds support for parsing and resolving dns-type `multiaddrs`, i.e. protocol schemes [`dns`, `dns4`, `dns6`, `dnsaddr`](https://github.com/multiformats/multiaddr/blob/b746a7d014e825221cc3aea6e57a92d78419990f/protocols.csv#L8-L11).

By default a `nwaku` node would attempt resolution against the Cloudflare name servers `1.1.1.1` and `1.0.0.1`, but this is configurable via the `dns-addrs-name-server` config option. DNS resolution can also be disabled completely by setting `dns-addrs` to `false`.

The next step would be to add a domain name config for a `nwaku` node which it can advertise. This is necessary, for example, to generate and sign a discoverable ENR containing a dns-type `multiaddr`.

cc @D4nte @dao

### CODE REVIEW

File: waku/v2/utils/peers.nim [hunk 1]
```diff
@@ -69,13 +69,13 @@ proc parseRemotePeerInfo*(address: string): RemotePeerInfo {.raises: [Defect, Va
...
-    ipPart, tcpPart, p2pPart, wsPart, wssPart: MultiAddress
+    nwPart, tcpPart, p2pPart, wsPart, wssPart: MultiAddress
 
   for addrPart in multiAddr.items():
     case addrPart[].protoName()[]
-    of "ip4", "ip6":
-      ipPart = addrPart.tryGet()
+    # All protocols listed here: https://github.com/multiformats/multiaddr/blob/b746a7d014e825221cc3aea6e57a92d78419990f/protocols.csv
+    of "ip4", "ip6", "dns", "dnsaddr", "dns4", "dns6":
+      nwPart = addrPart.tryGet()
```
Comment (oskarth) [inline waku/v2/utils/peers.nim:78]: What does `nw` stand for?

**Review by oskarth — APPROVED** [review]

**Comment by oskarth** [comment]
Nice!

File: waku/v2/utils/peers.nim (hunk 1 above)
Comment (jm-clius) [reply waku/v2/utils/peers.nim:78]: Network! :) With reference the network OSI layer. I think this part of the code probably needs a rethink, but I had to find some name to replace `ipPart`.

Comment (Menduist) [reply waku/v2/utils/peers.nim:78]: You can use `host`, that's the general name used in uri RFC etc

**Pull request merged**